src/main/java/teammates/logic/core/FeedbackResponsesLogic.java

```diff
@@ -838,4 +983,196 @@ private void addNewResponses(
 
         return viewableResponses;
     }
+    
+    public boolean isResponseValid(FeedbackResponseAttributes response,
+                                   FeedbackQuestionAttributes question) {
+        if (response == null || question == null) {
+            return false;
+        }
+        return response.isValid() && isResponseGiverAndReceiverValid(response, question);
+    }
+
+    public boolean isResponseGiverAndReceiverValid(FeedbackResponseAttributes response,
+                                                   FeedbackQuestionAttributes question) {
+        if (question == null || response == null) {
+            return false;
+        }
+
+        return isResponseGiverTypeMatchedToQuestionSetting(response, question) 
+               && isResponseRecipientTypeMatchedToQuestionSetting(response, question);
```

Comment:
@wkurniawan07 @damithc  If I verify a list of responses that are requested at a time instead of just one response, that will be faster?  That means, I only need to read database a few times for a list of responses rather than every response.